channeldb/migration14/migration.go

```diff
@@ -0,0 +1,90 @@
+package migration14
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/coreos/bbolt"
+)
+
+// MigrateCommitDiff migrates the CommitDiff struct on-disk for all active
+// channels to store the CommitSig with a length prefix, just as we do on the
+// wire. This ensures we'll be able to store any extra data appended to the
+// message, yet still serialize items after the message as well on disk.
+func MigrateCommitDiff(tx *bbolt.Tx) error {
+	openChanBucket := tx.Bucket(openChannelBucket)
```

Comment:
do we also need to migrate the historical channels bucket?

Reply:
Thought about that, but I don't think so, since we don't also migrate the `commitDiffKey` for them as there are no pending commitments given the channel is now closed.

Reply:
interesting, i was thinking we just copied over the state in-place, but if that's the case then perhaps not

Reply:
We _do_ copy over the state in-place, but that only the "core" channel state: https://github.com/lightningnetwork/lnd/blob/master/channeldb/channel.go#L2608 

The `commitDiffKey` is only updated here: https://github.com/lightningnetwork/lnd/blob/master/channeldb/channel.go#L1955 

Earlier in the transaction when we close the channels, we call [this method which clears out the commit diff state](https://github.com/lightningnetwork/lnd/blob/master/channeldb/channel.go#L2558).